modules/imgproc/include/opencv2/imgproc.hpp

```diff
@@ -4031,6 +4034,14 @@ CV_EXPORTS_W void findContours( InputArray image, OutputArrayOfArrays contours,
 CV_EXPORTS void findContours( InputArray image, OutputArrayOfArrays contours,
                               int mode, int method, Point offset = Point());
 
+#ifdef __OPENCV_BUILD
+CV_EXPORTS void findContours_old( InputArray _image, OutputArrayOfArrays _contours,
+                                  OutputArray _hierarchy, int mode, int method,
+                                  Point offset = Point() );
+CV_EXPORTS void findContours_old( InputArray image, OutputArrayOfArrays contours,
+                                  int mode, int method, Point offset = Point());
+#endif
```

Comment:
I propose to move it to `opencv2/imgproc/imgproc_c.h` or some `details` header. In case, if we need to presume the implementation, it's less painful for 4.x->5.x merges.
What about `findContoursLegacy` ?

Reply:
Moved to `imgproc/detail/legacy.hpp`, renamed to `findContours_legacy`.